pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -379,28 +332,32 @@ pub mod pallet {
 		/// message using the router, and calculates and returns the required
 		/// weight for these operations in the `DispatchResultWithPostInfo`.
 		fn process_outbound_message(
-			sender: T::AccountId,
+			sender: DomainAddress,
 			domain: Domain,
 			message: T::Message,
 		) -> (DispatchResult, Weight) {
-			let read_weight = T::DbWeight::get().reads(1);
+			let router_ids = T::RouterId::for_domain(domain);
 
-			let Some(router) = DomainRouters::<T>::get(domain) else {
-				return (Err(Error::<T>::RouterNotFound.into()), read_weight);
-			};
+			// TODO handle router ids logic
 
-			let (result, router_weight) = match router.send(sender, message.serialize()) {
-				Ok(dispatch_info) => (Ok(()), dispatch_info.actual_weight),
-				Err(e) => (Err(e.error), e.post_info.actual_weight),
-			};
+			let mut count = 0;
+			let bytes = message.serialize();
 
-			(result, router_weight.unwrap_or(read_weight))
+			for router_id in router_ids {
+				count += 1;
```

Comment:
Nit: Use safe math
```suggestion
				count.saturating_accrue(1);
```

Reply:
Thanks for the reminder! I think @cdamian already fixed in its PR, if not could you Cosmin?

Reply:
Sure.